apps/wallet/src/ui/app/components/ledger/ledgerErrors.ts

```diff
@@ -0,0 +1,46 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+interface LedgerTransportStatusError extends Error {
+    name: 'TransportStatusError';
+    statusCode: number;
+}
+
+export class LedgerConnectionFailedError extends Error {
+    constructor(message: string) {
+        super(message);
+        Object.setPrototypeOf(this, LedgerConnectionFailedError.prototype);
+    }
+}
+
+export class LedgerNoTransportMechanismError extends Error {
+    constructor(message: string) {
+        super(message);
+        Object.setPrototypeOf(this, LedgerNoTransportMechanismError.prototype);
+    }
+}
+
+export class LedgerDeviceNotFoundError extends Error {
+    constructor(message: string) {
+        super(message);
+        Object.setPrototypeOf(this, LedgerDeviceNotFoundError.prototype);
+    }
+}
+
+// Connection methods from the Ledger transport libraries don't throw well-structured
+// errors, so we can use this utility to form more explicit and structured errors
+export function convertErrorToLedgerConnectionFailedError(error: unknown) {
```

Comment:
this and `LedgerDeviceNotFoundError` are used in https://github.com/MystenLabs/sui/pull/9844